AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -541,7 +463,103 @@ - (void)performOperationForPartition:(NSString *)partition
+    completionHandler([[MSDocumentWrapper alloc] initWithError:serializationError documentId:documentId]);
+    return;
+  }
+  [self performCosmosDbOperationWithPartition:partition
+                                   documentId:documentId
+                                   httpMethod:kMSHttpMethodPost
+                                         body:body
+                            additionalHeaders:additionalHeaders
+                            completionHandler:^(NSData *_Nullable data, NSHTTPURLResponse *_Nullable __unused response,
+                                                NSError *_Nullable cosmosDbError) {
+                              // If not created.
+                              NSInteger errorCode = [MSDataSourceError errorCodeFromError:cosmosDbError];
+                              if (!data || (errorCode != MSACDocumentCreatedErrorCode && errorCode != MSACDocumentSucceededErrorCode)) {
+                                MSLogError([MSDataStore logTag], @"Not able to create/replace document: %@",
+                                           [cosmosDbError localizedDescription]);
+                                completionHandler([[MSDocumentWrapper alloc] initWithError:cosmosDbError documentId:documentId]);
+                                return;
+                              }
+
+                              // Deserialize.
+                              MSLogDebug([MSDataStore logTag], @"Document created/replaced with ID: %@", documentId);
+                              completionHandler([MSDocumentUtils documentWrapperFromData:data documentType:[document class]]);
+                            }];
+}
+
+- (void)deleteFromCosmosDbWithPartition:(NSString *)partition
+                             documentId:(NSString *)documentId
+                      completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+  [self performCosmosDbOperationWithPartition:partition
+                                   documentId:documentId
+                                   httpMethod:kMSHttpMethodDelete
+                                         // WIP: double check why we don't pass nil here
+                                         body:nil
+                            additionalHeaders:nil
+                            completionHandler:^(NSData *_Nullable __unused responseBody, NSHTTPURLResponse *_Nullable __unused response,
+                                                NSError *_Nullable cosmosDbError) {
+                              // Body returned from call (data) is empty.
+                              NSInteger httpStatusCode = [MSDataSourceError errorCodeFromError:cosmosDbError];
+                              if (httpStatusCode != MSHTTPCodesNo204NoContent) {
+                                MSLogError([MSDataStore logTag],
+                                           @"Not able to delete document. Error: %@; HTTP status code: %ld; "
+                                           @"Document: %@/%@",
```

Comment:
Seems this line can be combined with above.

Reply:
Leaving it out for now. We have a task to normalize logging all over the place.